src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingUseXmlReaderForDataTableReadXmlTests.cs

```diff
@@ -0,0 +1,397 @@
+            finally { }
+        }
+    }",
+                GetCA3075DataTableReadXmlCSharpResultAt(13, 17, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Data
+
+Class TestClass
+    Private Sub TestMethod()
+        Try
+            Dim src = """"
+            Dim dt As New DataTable()
+            dt.ReadXml(src)
+        Catch generatedExceptionName As Exception
+            Throw
+        Finally
+        End Try
+    End Sub
+End Class",
+                GetCA3075DataTableReadXmlBasicResultAt(9, 13, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseDataTableReadXmlInCatchBlockShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+   using System;
+   using System.Data;
+
+    class TestClass
+    {
+        private void TestMethod()
+        {
+            try { }
+            catch (Exception)
+            {
+                var src = """";
+                DataTable dt = new DataTable();
+                dt.ReadXml(src);
+            }
+            finally { }
+        }
+    }",
+                GetCA3075DataTableReadXmlCSharpResultAt(14, 17, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Data
+
+Class TestClass
+    Private Sub TestMethod()
+        Try
+        Catch generatedExceptionName As Exception
+            Dim src = """"
+            Dim dt As New DataTable()
+            dt.ReadXml(src)
+        Finally
+        End Try
+    End Sub
+End Class",
+                GetCA3075DataTableReadXmlBasicResultAt(10, 13, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseDataTableReadXmlInFinallyBlockShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Data;
+
+class TestClass
+{
+    private void TestMethod()
+    {
+        try { }
+        catch (Exception) { throw; }
+        finally
+        {
+            var src = """";
+            DataTable dt = new DataTable();
+            dt.ReadXml(src);
+        }
+    }
+}",
+                GetCA3075DataTableReadXmlCSharpResultAt(15, 13, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Data
+
+Class TestClass
+    Private Sub TestMethod()
+        Try
+        Catch generatedExceptionName As Exception
+            Throw
+        Finally
+            Dim src = """"
+            Dim dt As New DataTable()
+            dt.ReadXml(src)
+        End Try
+    End Sub
+End Class",
+                GetCA3075DataTableReadXmlBasicResultAt(12, 13, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseDataTableReadXmlInAsyncAwaitShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Threading.Tasks;
+using System.Data;
+
+class TestClass
+{
+    private async Task TestMethod()
+    {
+        await Task.Run(() => {
+            var src = """";
+            DataTable dt = new DataTable();
+            dt.ReadXml(src);
+        });
+    }
+
```

Comment:
Because the test has the same effect with or without those four lines; in other words, we have four lines of code that serve no purpose. Lines that have to be inspected by a human, stored, and compiled into object code. We should eliminate useless code wherever we find it.